### ISSUE TITLE
Discussion: Trading value of filthy clothing?

### ISSUE THREAD

**Issue by pjf** [CleverRaven/Cataclysm-DDA#49469]
I'm back at poking around the trading code again, and it's bothering me that filthy items seem to have the same value as non-filthy items. Especially since filthy items have lots of disadvantages (infection risk, can't be used for crafting, unsightly colour in inventory, etc). I feel that merchants and survivors should refuse or pay considerably less for filthy things.

I've been poking around the trading and item price code, and would like to fix this, but I see a few options available, such as:

1. Disallow filthy trading entirely.
1. Make filthy items worth a percentage less than clean items. (Eg: half-price)
2. Reduce the value of filthy items by a set amount, representing the labour and materials needed to clean them.
3. Something more complex (based on NPC flags, etc)

Of all of these, I like reducing the value by a set amount, as it means:

- Cheap, common items lose most or all of their value. Merchants don't want your stack of twenty torn and filthy t-shirts
- Valuable items are still valuable. Good armour is worth it, even if it needs a wash.
- Minimal code changes (100% confined to `item::price()`).

With a set price reduction of `-100` for filthy items, for example:

- Common clothes (including hoodies) become worthless
- Midline clothes (eg: leather vests) are about half-price
- Valuable clothes and above (good boots, real armour, etc) retain most of their value

Thoughts and feedback?

**tl;dr: I'd like to add a fixed price malus to filthy items in trade**

**Comment by PatrikLundell** [comment]
If the merchant has access to a washing machine, the cleaning cost reduction would depend on the item volume on their end (you can stuff more small things into the machine than large ones). If they don't have access to cheap cleaning they shouldn't be interested at all, unless it's something very valuable.

However, a flat reduction seems to be reasonable, and sounds, from a distance, like a reasonably easy thing to implement.

**Comment by faefux** [comment]
> If the merchant has access to a washing machine, the cleaning cost reduction would depend on the item volume on their end (you can stuff more small things into the machine than large ones). If they don't have access to cheap cleaning they shouldn't be interested at all, unless it's something very valuable.
> 
> However, a flat reduction seems to be reasonable, and sounds, from a distance, like a reasonably easy thing to implement.

I don't like the washing machine argument, because... well, we have tubs, sticks and washboards... (And have you seen the fabriaction level needed for a washboard?! Absurdly low!  )

**Comment by PatrikLundell** [comment]
Do you know how much work it is to wash things by hand? It's not a coincidence that washer woman was a profession before washing machines became commonly available. The key here is "cheap" (although it wasn't a well paid prestigious job, they still had to be paid).

**Comment by faefux** [comment]
> Do you know how much work it is to wash things by hand? It's not a coincidence that washer woman was a profession before washing machines became commonly available. The key here is "cheap" (although it wasn't a well paid prestigious job, they still had to be paid).

Unless I grossly misunderstood, We could say also that washing machine is not cheap to run, for many more expensive than the physical labour.  I suppose it is all theoretical regardless, Overall though I think more granularity in trade like you are suggesting would be real neat.

**Comment by LeahLuong** [comment]
@pjf Paul, still thinking about it??
